bot/exts/evergreen/avatar_modification/avatar_modify.py

```diff
@@ -308,6 +313,49 @@ async def spookyavatar(self, ctx: commands.Context, member: discord.Member = Non
 
             await ctx.send(file=file, embed=embed)
 
+    @avatar_modify.command(name='mosaic', root_aliases=('mosaic',))
+    async def mosaic_command(self, ctx: commands.Context, squares: int = 16) -> None:
+        """Splits your avatar into x squares, randomizes them and stitches them back into a new image!"""
+        async with ctx.typing():
+            if squares < 1:
+                raise commands.BadArgument("Squares must be a positive number")
+
+            if not math.sqrt(squares).is_integer():
+                raise commands.BadArgument("Squares must be a perfect square")
+
+            if squares > MAX_SQUARES:
+                raise commands.BadArgument(f"Number of squares cannot be higher than {MAX_SQUARES:,}.")
+
+            file_name = file_safe_name("mosaic_avatar", ctx.author.display_name)
+            img_bytes = await ctx.author.avatar_url.read()
```

Comment:
i'd suggest using the `self._fetch_member()` function here, to get the a new member object, since we don't have the member intent enabled, the cache doesn't have an updated version of the avatar.

You can use this new member object in the rest of this function then, replacing ctx.author.

Reply:
Resolved in https://github.com/python-discord/sir-lancebot/pull/716/commits/544a05b00758583f4594569e032c9e661406a72f.